crates/iceberg-catalog/src/api/management/v1/warehouse.rs

```diff
@@ -219,21 +257,15 @@ pub trait Service<C: Catalog, A: AuthZHandler, S: SecretStore> {
         request_metadata: RequestMetadata,
     ) -> Result<ListProjectsResponse> {
         // ------------------- AuthZ -------------------
-        let projects = A::check_list_projects(&request_metadata, context.v1_state.auth).await?;
+        let authorizer = context.v1_state.authz;
+        let projects = authorizer.list_projects(&request_metadata).await?;
 
         // ------------------- Business Logic -------------------
-        if let Some(projects) = projects {
-            return Ok(ListProjectsResponse {
-                projects: projects
-                    .into_iter()
-                    .map(|project_id| ProjectResponse {
-                        project_id: *project_id,
-                    })
-                    .collect(),
-            });
-        }
-
-        let projects = C::list_projects(context.v1_state.catalog).await?;
+        let project_id_filter = match projects {
+            AuthZListProjectsResponse::All => None,
+            AuthZListProjectsResponse::Projects(projects) => Some(projects),
+        };
+        let projects = C::list_projects(project_id_filter, context.v1_state.catalog).await?;
```

Comment:
should add an issue to add pagination

Reply:
https://github.com/hansetag/iceberg-catalog/issues/397